hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/validation/TestRequestValidations.java

```diff
@@ -0,0 +1,422 @@
+
+    validationListener.assertNumOfEvents(3);
+    validationListener.assertAllCalled(
+//        "unconditionalPostProcessCreateKeyValidator",
+        "preFinalizePostProcessCreateKeyValidator",
+        "oldClientPostProcessCreateKeyValidator",
+        "oldClientPostProcessCreateKeyValidator2");
+  }
+
+  private RequestValidations loadValidations(ValidationContext ctx) {
+    return new RequestValidations()
+        .fromPackage(PACKAGE)
+        .withinContext(ctx)
+        .load();
+  }
+
+  private RequestValidations loadEmptyValidations(ValidationContext ctx) {
+    return new RequestValidations()
+        .fromPackage(PACKAGE_WO_VALIDATORS)
+        .withinContext(ctx)
+        .load();
+  }
+
+  private int olderClientVersion() {
+    return ClientVersions.CURRENT_VERSION - 1;
+  }
+
+  private int currentClientVersion() {
+    return ClientVersions.CURRENT_VERSION;
+  }
+
+/*  private int newerClientVersion() {
+    return ClientVersions.CURRENT_VERSION + 1;
+  }*/
+
+  private OMRequest aCreateKeyRequest(int clientVersion) {
+    return aRequest(CreateKey, clientVersion);
+  }
+
+  private OMRequest aDeleteKeysRequest(int clientVersion) {
+    return aRequest(DeleteKeys, clientVersion);
+  }
+
+  private OMRequest aRenameKeyRequest(int clientVersion) {
+    return aRequest(RenameKey, clientVersion);
+  }
+
+  private OMRequest aRequest(Type type, int clientVersion) {
+    return OMRequest.newBuilder()
+        .setVersion(clientVersion)
+        .setCmdType(type)
+        .setClientId("TestClient")
+        .build();
+  }
+
+  private OMResponse aCreateKeyResponse() {
+    return aResponse(CreateKey);
+  }
+
+  private OMResponse aDeleteKeysResponse() {
+    return aResponse(DeleteKeys);
+  }
+
+  private OMResponse aRenameKeyResponse() {
+    return aResponse(RenameKey);
+  }
+
+  private OMResponse aResponse(Type type) {
+    return OMResponse.newBuilder()
+        .setCmdType(type)
+        .setStatus(OK)
+        .build();
+  }
+
+  private LayoutVersionManager aFinalizedVersionManager() {
+    LayoutVersionManager vm = mock(LayoutVersionManager.class);
+    when(vm.needsFinalization()).thenReturn(false);
+    return vm;
+  }
+
+  private LayoutVersionManager anUnfinalizedVersionManager() {
+    LayoutVersionManager vm = mock(LayoutVersionManager.class);
+    when(vm.needsFinalization()).thenReturn(true);
+    return vm;
+  }
+
+  private static class ValidationListenerImpl implements ValidationListener {
+    private List<String> calledMethods = new ArrayList<>();
+
+    @Override
+    public void validationCalled(String calledMethodName) {
+      calledMethods.add(calledMethodName);
+    }
+
+    public void attach() {
+      GeneralValidatorsForTesting.addListener(this);
+    }
+
+    public void detach() {
+      GeneralValidatorsForTesting.removeListener(this);
+      reset();
+    }
+
+    public void reset() {
+      calledMethods = new ArrayList<>();
+    }
+
+    public void assertCalled(String... methodNames) {
```

Comment:
Are there tests right now where this is used but `assertAllCalled` would not work in its place?

Reply:
The two method serves a different purpose, this is to check if the methods were called, but does not bother if other methods were also called. This is because I had some troubles to figure out how to restrict Reflections to really just one package instead of the jar file(s) that contains the package.
I switched to allcalled, and renamed that method to better represent what it does to assertExactListOfValidatorsCalled(String...)